packages/wrangler/src/hyperdrive/client.ts

```diff
@@ -9,20 +9,46 @@ export type HyperdriveConfig = {
 	caching: CachingOptions;
 };
 
-export type PublicOrigin = {
-	host?: string;
-	port?: number;
-	scheme?: string;
-	database?: string;
-	user?: string;
-	access_client_id?: string;
+export type OriginCommon = {
+	host: string;
+	scheme: string;
+	database: string;
+	user: string;
 };
 
-export type OriginWithSecrets = PublicOrigin & {
-	password?: string;
-	access_client_secret?: string;
+export type OriginCommonWithSecrets = OriginCommon & {
+	password: string;
 };
 
+export type OriginHoA = OriginCommon & {
+	access_client_id: string;
+	access_client_secret?: never;
```

Comment:
Having the base OriginHoA include the secrets, when we have a WithSecrets version, seems wrong to me?

Reply:
`OriginHoa` is actually enforcing that the secret is not present here, this type is saying "access_client_secret" must never be set.